uqcsbot/utils/uq_course_utils.py

```diff
@@ -16,12 +16,87 @@
 OFFERING_PARAMETER = "offer"
 
 
+class Offering:
+    """
+    A semester, campus and mode (e.g. Internal) that many courses occur within
+    """
+
+    CampusType = Literal["St Lucia", "Gatton", "Herston"]
+    # The codes used internally within UQ systems
+    campus_codes: Dict[CampusType, str] = {
+        "St Lucia": "STLUC",
+        "Gatton": "GATTN",
+        "Herston": "HERST",
+    }
+
+    ModeType = Literal["Internal", "External", "Flexible Delivery"]
+    # The codes used internally within UQ systems
+    mode_codes: Dict[ModeType, str] = {
+        "Internal": "IN",
+        "External": "EX",
+        "Flexible Delivery": "FD",
+    }
+
+    SemesterType = Literal["1", "2", "Summer"]
+    semester_codes: Dict[SemesterType, int] = {"1": 1, "2": 2, "Summer": 3}
+
+    semester: SemesterType
+    campus: CampusType
+    mode: ModeType
+
+    def __init__(
+        self,
+        semester: Optional[SemesterType],
+        campus: CampusType = "St Lucia",
+        mode: ModeType = "Internal",
+    ):
+        """
+        semester defaults to the current semester if None
+        """
+        if semester is not None:
+            self.semester = semester
+        else:
+            self.semester = "1" if datetime.today().month <= 6 else "2"
```

Comment:
This doesn't appear to handle defaulting to summer sem from (approximately) November to February.